newIDE/app/src/MainFrame/EditorContainers/StartPage/Carousel.js

```diff
@@ -0,0 +1,338 @@
+    [widthUnit, itemsToDisplay]
+  );
+
+  const roundScroll = React.useCallback(
+    (value: number): number => {
+      return Math.round(value / widthUnit) * widthUnit;
+    },
+    [widthUnit]
+  );
+
+  const onClickArrow = (direction: 'left' | 'right') =>
+    React.useCallback(
+      (): void => {
+        const scrollViewElement = scrollView.current;
+        if (!scrollViewElement) return;
+        const newScrollPosition = computeScroll(direction, scrollViewElement);
+
+        scrollViewElement.scrollTo({
+          left: newScrollPosition,
+          behavior: 'smooth',
+        });
+      },
+      [computeScroll]
+    );
+
+  const handleScroll = React.useCallback(() => {
+    const scrollViewElement = scrollView.current;
+    if (!scrollViewElement) return;
+
+    if (!shouldDisplayLeftArrow)
+      setShouldDisplayLeftArrow(scrollViewElement.scrollLeft !== 0);
+  });
+  const handleScrollEnd = React.useCallback(
+    () => {
+      const scrollViewElement = scrollView.current;
+      if (!scrollViewElement) return;
+
+      scrollViewElement.scrollTo({
+        left: roundScroll(scrollViewElement.scrollLeft),
+        behavior: 'smooth',
+      });
+    },
+    [roundScroll]
+  );
+
+  React.useEffect(() => {
+    const scrollViewElement = scrollView.current;
+    if (!scrollViewElement) return;
+
+    scrollViewElement.addEventListener('scroll', handleScroll);
+    scrollViewElement.addEventListener('touchend', handleScrollEnd);
+    scrollViewElement.addEventListener('touchleave', handleScrollEnd);
+  }, []);
+
+  return (
+    <Line noMargin expand>
+      <div
+        style={{ ...styles.leftArrowContainer, width: arrowWidth }}
+        onClick={onClickArrow('left')}
+      >
+        {!!items && shouldDisplayLeftArrow && <ArrowBackIos />}
+      </div>
+      <div style={{ width: `calc(100% - ${arrowWidth}px)` }}>
+        <Text size="title">{title}</Text>
+        <GridList
+          cols={itemsToDisplay.length}
+          cellHeight={cellHeight}
+          spacing={cellSpacing}
+          style={styles.gridList}
+          ref={scrollView}
+        >
+          {itemsToDisplay.map(item => (
+            <GridListTile key={computeItemKey(item)}>
+              {renderThumbnail(item)}
+              {renderItemTitle(item)}
+            </GridListTile>
+          ))}
+          {!!items && (
+            <div
+              style={{
+                ...styles.rightArrowContainer,
+                height: cellHeight - 12,
+                width: arrowWidth,
+              }}
+              onClick={onClickArrow('right')}
+            >
+              <ArrowForwardIos htmlColor="white" />
+            </div>
+          )}
+        </GridList>
+      </div>
+    </Line>
+  );
+};
+
+const extractVideoIdFromYoutubeLink = (link: string): ?string => {
+  const url = new URL(link);
+  const params = url.searchParams;
+  return params.get('v') || null;
+};
+
+const constructImageLinkFromYoutubeLink = (link: string): string => {
```

Comment:
This should be returned by the API in the future I guess? cc @ClementPasteau

Reply:
When discussing it, we were not sure, but I have not seen what has been made in the API side. This logic can be used as a default though.

Reply:
Ok we'll see then, a first draft of the API should be done now, probably the API should indicate the image to use (I think there will be some logic so that we always host the images). I'll let you double check! :)